IntelliTect.TestTools.WindowsTestWrapper/ExampleProjects/SimpleWpfApp.AutomatedUiTests/Windows/SimpleWpfAppWindow.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using IntelliTect.TestTools.WindowsTestWrapper;
+using Microsoft.VisualStudio.TestTools.UITesting;
+using Microsoft.VisualStudio.TestTools.UITesting.WpfControls;
+
+
+namespace SimpleWpfApp.AutomatedUiTests.Windows
+{
+    public class SimpleWpfAppWindow : DesktopControls
+    {
+        public override string AutWindowTitle => Aut?.Title;
+
+        //This will vary depending on if the playback engine is using MSAA or UIA.
+        //More application support MSAA and so is a safer bet at this time.
+        public override string WindowClassName => "HwndWrapper";
+
+        public static void LaunchApplicationUnderTest()
+        {
+            LaunchApplication( Path.GetFullPath(Path.Combine(Environment.CurrentDirectory, @"..\..\..\ExampleProjects\SimpleWpfApp\bin\Debug\SimpleWpfApp.exe")) );
+        }
+
+        public WpfEdit ListEntry
+        {
+            get { return FindWpfControlByAutomationId( "autoTextInput", c => new WpfEdit( c ) ); }
+        }
+
+        public UITestControl SaveButton
+        {
+            get { return BuildControlHeirarchy( SearchTypes.AutomationId, "autoButtonSave" ); }
+        }
+
+        public UITestControl ExpandListItem
+        {
+            get { return BuildControlHeirarchy( SearchTypes.AutomationId, "autoExpanderList", "HeaderSite" ); }
+        }
+
+        private UITestControl ListItems
+        {
+            get { return BuildControlHeirarchy( SearchTypes.AutomationId, "autoExpanderList", "autoListBoxMyList" ); }
+        }
+
+        public IEnumerable<WpfListItem> ListItemsList
+        {
+            get { return GetListOfControlsByType( c => new WpfListItem( c ), ListItems ); }
+        }
+
+        public bool CheckAllListItems(params string[] expectedItems)
+        {
+            if ( expectedItems == null )
+                throw new ArgumentNullException( nameof( expectedItems ) );
+
+            var existingListItems = ListItemsList;
+
+            if ( expectedItems.Length != ListItemsList.Count() )
```

Comment:
There is an additional code smell here because it looks like you want to iterate over the enumerable twice. You probably want to do something like
```C#
List<WpfListItem> existingListItems = ListItemsList?.ToList();
if (expectedItems.Length != existingListItems?.Count)
```